flake.nix

```
@@ -27,8 +27,84 @@
                         base
                         wai
                         text
+
+                        # Uncomment on local development for testing
+                        # hspec
                     ];
                 };
+
+                # Custom configuration that will start with `devenv up`
+                devenv.shells.default = {
+                    # Start Mailhog on local development to catch outgoing emails
+                    services.mailhog.enable = true;
+
+                    # Custom processes that don't appear in https://devenv.sh/reference/options/
+                    processes = {
+                        # Uncomment if you use tailwindcss.
+                        # tailwind.exec = "tailwindcss -c tailwind/tailwind.config.js -i ./tailwind/app.css -o static/app.css --watch=always";
+                    };
+                };
+            };
+
+            # Adding the new NixOS configuration for "qa"
+            # Used to deploy the IHP application to AWS.
+            #
+            # Change the `CHANGE-ME` to your correct config.
+            flake.nixosConfigurations."qa" = nixpkgs.lib.nixosSystem {
+                system = "x86_64-linux";
+                specialArgs = inputs;
+                modules = [
+                    "${nixpkgs}/nixos/modules/virtualisation/amazon-image.nix"
+                    ihp.nixosModules.appWithPostgres
+                    ({ lib, pkgs, ... }: {
+
+                        networking.firewall = {
+                            enable = true;
+                            allowedTCPPorts = [ 22 80 443 8000 ];
```

```suggestion
                            allowedTCPPorts = [ 22 80 443 ];
```

8000 shouldn't be exposed